src/Server/Auth.fs

```diff
@@ -1,15 +1,13 @@
 /// Login web part and functions for API web part request authorisation with JWT.
 module ServerCode.Auth
 
-open Suave
-open Suave.RequestErrors
+open Giraffe
+open RequestErrors
+open Microsoft.AspNetCore.Http
 
 /// Login web part that authenticates a user and returns a token in the HTTP body.
-let login (ctx: HttpContext) = async {
-    let login = 
-        ctx.request.rawForm 
-        |> System.Text.Encoding.UTF8.GetString
-        |> FableJson.ofJson<Domain.Login>
+let login next (ctx: HttpContext) = task {
+    let! login = FableJson.getJsonFromCtx<Domain.Login> ctx
```

Comment:
We're using the following helper (note it's asynchronous):

```fsharp
  let rawForm (req: HttpRequest) =
      use reader = new StreamReader(req.Body, System.Text.Encoding.UTF8)
      reader.ReadToEndAsync()
```
And then:

```fsharp
    let! reqData = Http.rawForm ctx.Request
    let login = FableJson.ofJson<Domain.Login> reqData
```

Reply:
it's same thing, right? mine just has it in one function

Reply:
Yeah, I just mentioned it to have a way to get the body contents in a similar way as in Suave.

Reply:
just found   
    member this.ReadBodyFromRequestAsync() =
        let body = this.Request.Body
        use reader = new StreamReader(body, true)
        reader.ReadToEndAsync()